autospec/specfiles.py

```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
                 file2 = file.replace("/", "_")
                 self._write_strip("cp " + file + " %{buildroot}/usr/share/package-licenses/" + self.name + "/" + file2 + "\n")
 
+    def write_profile_payload(self, pattern=None):
+        """Write the profile_payload specified for this package."""
+        if not pattern:
+            return
+        if not config.profile_payload:
+            return
+        if config.profile_payload and not config.profile_payload[0]:
+            return
+        use_subdir = True
+        init = ""
+        post = ""
+        if pattern == "configure":
+            init = "{0}%configure {1} {2} {3}".format(
+                   self.get_profile_generate_flags(),
+                   self.disable_static,
+                   config.extra_configure,
+                   config.extra_configure64)
+        elif pattern == "configure_ac":
+            init = "{0}%reconfigure {1} {2} {3}".format(
+                   self.get_profile_generate_flags(),
+                   self.disable_static,
+                   config.extra_configure,
+                   config.extra_configure64)
+        elif pattern == "autogen":
+            init = "{0}%autogen {1} {2} {3}".format(
+                   self.get_profile_generate_flags(),
+                   self.disable_static,
+                   config.extra_configure,
+                   config.extra_configure64)
+        elif pattern == "cmake":
+            use_subdir = False
+            init = "{0}".format(self.get_profile_generate_flags())
+            post = "{0}".format(self.get_profile_use_flags())
+        else:
+            print_warning(f"profile_payload not supported for {pattern} pattern")
```

Comment:
maybe this should be a hard error for testing

Reply:
I removed this block in v2